samples/bookinfo/src/details/details.rb

```diff
@@ -79,7 +79,7 @@ def get_book_details(id, headers)
 
 def fetch_details_from_external_service(isbn, id, headers)
     uri = URI.parse('https://www.googleapis.com/books/v1/volumes?q=isbn:' + isbn)
-    http = Net::HTTP.new(uri.host, uri.port)
+    http = Net::HTTP.new(uri.host, ENV['DO_NOT_ENCRYPT'] === 'true' ? 80:443)
```

Comment:
This doesn't make sense to me.  The URL scheme is hard coded to be https.  So why would you create the option to send a request to port 80 for a https URL?

Reply:
@duderino This example is used in https://istio.io/blog/2018/egress-https/#tls-origination-by-istio. The app can issue an HTTP request and Istio will perform TLS origination for it. I should have documented it in the first comment of the PR.